meinberlin/react/kiezradar/SearchProfiles.jsx

```diff
@@ -0,0 +1,294 @@
+      })
+
+      if (!response.ok) {
+        throw new Error(errorDeleteSearchProfilesText)
+      }
+
+      onDelete(profile.id)
+    } catch (err) {
+      setError(err.message)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    setLoading(true)
+    setError(null)
+
+    try {
+      const response = await fetch(apiUrl + profile.id + '/', {
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8',
+          'X-CSRFToken': cookie.get('csrftoken')
+        },
+        method: 'PATCH',
+        body: JSON.stringify({ name: e.target.elements.name.value })
+      })
+
+      if (!response.ok) {
+        throw new Error(errorUpdateSearchProfilesText)
+      }
+
+      const data = await response.json()
+      setProfile(data)
+    } catch (err) {
+      setError(err.message)
+    } finally {
+      setIsEditing(false)
+      setLoading(false)
+    }
+  }
+
+  return (
+    <div className="search-profile">
+      <div className="search-profile__header">
+        <h3 className="search-profile__title">{profile.name}</h3>
+        {!isEditing && (
+          <div className="search-profile__header-buttons">
+            <Buttons
+              onEdit={() => setIsEditing(true)}
+              onDelete={handleDelete}
+              loading={loading}
+            />
+          </div>
+        )}
+      </div>
+      {error && <div className="search-profile__error">{errorText + ': ' + error}</div>}
+      {isEditing && (
+        <form className="form--base panel--heavy search-profile__form" onSubmit={handleSubmit}>
+          <div className="form-group">
+            <label htmlFor="name">{renameSearchProfileText}</label>
+            <input id="name" name="name" type="text" required />
+          </div>
+          <div className="form-actions">
+            <div className="form-actions__left">
+              <button className="link" onClick={() => setIsEditing(false)}>
+                {cancelText}
+              </button>
+            </div>
+            <div className="form-actions__right">
+              <button
+                className="button"
+                type="submit"
+                disabled={loading}
+              >
+                {loading ? savingText + '...' : saveText}
+              </button>
+            </div>
+          </div>
+        </form>
+      )}
+      <div className="search-profile__footer">
+        <a href={planListUrl + '?search-profile=' + profile.id} className="button button--light search-profile__view-projects">
+          {viewProjectsText}
+        </a>
+        {!isEditing && (
+          <div className="search-profile__footer-buttons">
+            <Buttons
+              onEdit={() => setIsEditing(true)}
+              onDelete={handleDelete}
+              loading={loading}
+            />
+          </div>
+        )}
+      </div>
+    </div>
+  )
+}
+
+function Buttons ({ onEdit, onDelete, loading }) {
```

Comment:
Could you give this some more descriptive name after you split the components into separate files? Otherwise this would probably show up for a lot of autocompletes in IDE